### ISSUE TITLE
Syntax error in nested dictionary value reported on top line instead of value's line

### ISSUE THREAD

**Issue by salty-horse** [PyCQA/pyflakes#731]
Using Pyflakes 2.5.0 with Python 3.10.4

Given this file with a syntax error (`===`), pyflakes gives helpful output:
```python
{
    'b': {
        'c': 1,
        'd': 1 === 2,
    },
}
```
```
test.py:4:18: invalid syntax
        'd': 1 === 2,
                 ^
```

However, if the inner dictionary is preceded by another field, the error message points at the top of the dictionary, rather than the offending line:

```python
{
    'a': 1, # New field
    'b': {
        'c': 1,
        'd': 1 === 2,
    },
}
```
```
test.py:3:8: expression expected after dictionary key and ':'
    'b': {
       ^
```

Is it possible to improve the detection of this syntax error?

**Comment by asottile** [comment]
pyflakes is just disposing what cpython is reporting

**Comment by salty-horse** [comment]
Thanks. This is fixed in CPython 3.11 :)
